File: lib/services/auth_service.dart
import '../models/user_model.dart';

class AuthService {
  static User? _currentUser;
  static bool _isAuthenticated = false;

  // Get current user
  User? getCurrentUser() => _currentUser;

  // Check if user is authenticated
  bool isAuthenticated() => _isAuthenticated;

  // Login
  Future<bool> login(String email, String password) async {
    await Future.delayed(const Duration(milliseconds: 800));

    // Validation
    if (email.isEmpty || password.isEmpty) {
      return false;
    }

    if (!email.contains('@')) {
      return false;
    }

    if (password.length < 6) {
      return false;
    }

    // Simulate successful login
    _currentUser = User(
      id: '${DateTime.now().millisecondsSinceEpoch}',
      email: email,
      firstName: email.split('@')[0],
      lastName: 'Student',
      phoneNumber: '+1234567890',
      university: 'Tech University',
      createdAt: DateTime.now(),
      isEmailVerified: true,
    );
    _isAuthenticated = true;
    return true;
  }

  // Register
  Future<bool> register({
    required String email,
    required String password,
    required String confirmPassword,
    required String firstName,
    required String lastName,
    required String phoneNumber,
    required String university,
  }) async {
    await Future.delayed(const Duration(milliseconds: 800));

    // Validation
    if (email.isEmpty ||
        password.isEmpty ||
        firstName.isEmpty ||
        lastName.isEmpty) {
      return false;
    }

    if (!email.contains('@')) {
      return false;
    }

    if (password.length < 6) {
      return false;
    }

    if (password != confirmPassword) {
      return false;
    }

    if (phoneNumber.length < 10) {
      return false;
    }

    // Simulate successful registration
    _currentUser = User(
      id: '${DateTime.now().millisecondsSinceEpoch}',
      email: email,
      firstName: firstName,
      lastName: lastName,
      phoneNumber: phoneNumber,
      university: university,
      createdAt: DateTime.now(),
      isEmailVerified: false,
    );
    _isAuthenticated = true;
    return true;
  }

  // Logout
  Future<void> logout() async {
    await Future.delayed(const Duration(milliseconds: 500));
    _currentUser = null;
    _isAuthenticated = false;
  }

  // Update profile
  Future<bool> updateProfile({
    String? firstName,
    String? lastName,
    String? phoneNumber,
    String? university,
  }) async {
    await Future.delayed(const Duration(milliseconds: 500));

    if (_currentUser == null) return false;

    _currentUser = _currentUser!.copyWith(
      firstName: firstName ?? _currentUser!.firstName,
      lastName: lastName ?? _currentUser!.lastName,
      phoneNumber: phoneNumber ?? _currentUser!.phoneNumber,
      university: university ?? _currentUser!.university,
    );
    return true;
  }

  // Verify email
  Future<bool> verifyEmail(String code) async {
    await Future.delayed(const Duration(milliseconds: 600));

    if (code == '123456') {
      if (_currentUser != null) {
        _currentUser = _currentUser!.copyWith(isEmailVerified: true);
      }
      return true;
    }
    return false;
  }

  // Reset password
  Future<bool> resetPassword(String email) async {
    await Future.delayed(const Duration(milliseconds: 600));
    return email.contains('@');
  }
}

File: services/authService.ts
import { User } from "@/models/user";

let currentUser: User | null = null;
let isAuthenticated = false;

const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));

export const authService = {
  getCurrentUser: () => currentUser,
  isAuthenticated: () => isAuthenticated,

  async login(email: string, password: string): Promise<boolean> {
    await delay(800);
    if (!email || !password || !email.includes("@") || password.length < 6) {
      return false;
    }
    const namePart = email.split("@")[0];
    currentUser = {
      id: `${Date.now()}`,
      email,
      firstName: namePart,
      lastName: "Student",
      phoneNumber: "+254712345678",
      university: "University of Nairobi",
      createdAt: new Date(),
      isEmailVerified: true,
    };
    isAuthenticated = true;
    return true;
  },

  async register(params: {
    email: string;
    password: string;
    confirmPassword: string;
    firstName: string;
    lastName: string;
    phoneNumber: string;
    university: string;
  }): Promise<boolean> {
    await delay(800);
    const { email, password, confirmPassword, firstName, lastName, phoneNumber, university } = params;
    if (!email || !password || !firstName || !lastName) return false;
    if (!email.includes("@") || password.length < 6) return false;
    if (password !== confirmPassword) return false;
    if (phoneNumber.replace(/\D/g, "").length < 10) return false;

    currentUser = {
      id: `${Date.now()}`,
      email,
      firstName,
      lastName,
      phoneNumber,
      university,
      createdAt: new Date(),
      isEmailVerified: false,
    };
    isAuthenticated = true;
    return true;
  },

  async logout(): Promise<void> {
    await delay(500);
    currentUser = null;
    isAuthenticated = false;
  },

  async resetPassword(email: string): Promise<boolean> {
    await delay(600);
    return email.includes("@");
  },

  async updateProfile(updates: Partial<Pick<User, "firstName" | "lastName" | "phoneNumber" | "university">>): Promise<boolean> {
    await delay(500);
    if (!currentUser) return false;
    currentUser = { ...currentUser, ...updates };
    return true;
  },
};
